test/HealthChecks.SendGrid.Tests/DependencyInjection/RegistrationTests.cs

```diff
@@ -19,6 +19,30 @@ public void add_health_check_when_properly_configured()
         check.ShouldBeOfType<SendGridHealthCheck>();
     }
 
+    private class SendGridOptions
+    {
+        public string ApiKey { get; set; } = default!;
+    }
+
+    [Fact]
+    public void add_health_check_from_DI_when_properly_configured()
+    {
+        var services = new ServiceCollection();
+
+        services.AddOptions<SendGridOptions>().Configure(options => options.ApiKey = "my_api_key");
+        services.AddHealthChecks()
+            .AddSendGrid(sp => sp.GetRequiredService<IOptions<SendGridOptions>>().Value.ApiKey);
+
+        using var serviceProvider = services.BuildServiceProvider();
+        var options = serviceProvider.GetRequiredService<IOptions<HealthCheckServiceOptions>>();
+
+        var registration = options.Value.Registrations.First();
+        var check = registration.Factory(serviceProvider);
+
+        registration.Name.ShouldBe("sendgrid");
+        check.ShouldBeOfType<SendGridHealthCheck>();
```

Comment:
Do I read this correctly that this test is not veryfing if value `my_api_key` has been actually used by the health check?

Reply:
i used the same assertion code that was acceptable for the previous test.